portal-impl/test/unit/com/liferay/portal/util/FileImplTest.java

```diff
@@ -223,6 +236,134 @@ public void testStripSuffixWhenFileNameHasParenthesisAtStart() {
 			"()test.jsp", _fileImpl.stripParentheticalSuffix("()test.jsp"));
 	}
 
+	@Test
+	public void testUnzip() throws Exception {
+		_setUpForUnzipTests();
+
+		File file = _createZipFile(
+			"test.zip",
+			HashMapBuilder.put(
+				"zip/test/directory/", () -> (String)null
+			).put(
+				"zip/test/entry/entry.txt", "Test String"
+			).build());
+
+		_fileImpl.unzip(file, _tempFolder);
+
+		_assertExists("zip/test/directory");
+		_assertExists("zip/test/entry/entry.txt");
+
+		Assert.assertTrue(
+			_canFileReadWrite(_getTempTestFilePath("zip/test/directory")));
+
+		Assert.assertTrue(
+			_canFileReadWrite(
+				_getTempTestFilePath("zip/test/entry/entry.txt")));
+		_tearDown();
+	}
+
+	@Test
+	public void testUnzipZipSlipVulnerable() throws Exception {
```

Comment:
@jeffwu0724 For the vulnerable test, I think you should add assertion for the warning log output. On one hand this ensures the log is printed as expected, on the other hand it can avoid the warning to be printed to the console.

For example, when i run the unit test locally I see the following output:

```
    [junit] Running com.liferay.portal.util.FileImplTest
    [junit] Testsuite: com.liferay.portal.util.FileImplTest
    [junit] SLF4J: Class path contains multiple SLF4J bindings.
    [junit] SLF4J: Found binding in [file:/home/repo/liferay/portal-ce/util-slf4j/classes/org/slf4j/impl/StaticLoggerBinder.class]
    [junit] SLF4J: Found binding in [jar:file:/home/repo/liferay/portal-ce/lib/development/slf4j-simple.jar!/org/slf4j/impl/StaticLoggerBinder.class]
    [junit] SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
    [junit] Loading file:/home/repo/liferay/portal-ce/portal-impl/classes/system.properties
    [junit] SLF4J: Actual binding is of type [com.liferay.util.slf4j.LiferayLoggerFactory]
    [junit] Loading file:/home/repo/liferay/portal-ce/portal-impl/classes/portal.properties
    [junit] Loading file:/home/repo/liferay/portal-ce/portal-impl/test-classes/unit/portal-test.properties
    [junit] Tests run: 28, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.264 sec
    [junit] Tests run: 28, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.264 sec
    [junit] 
    [junit] 六月 07, 2022 5:42:09 上午 com.liferay.portal.kernel.log.Jdk14LogImpl warn
    [junit] 警告: Entry is outside of the target dir: ../../../../../../bad.txt
[stopwatch] [test.cmd: 0.807 sec]
```

The "警告" line is the logger's warning output, localized by my OS configuration.

You can take https://github.com/liferay-core-infra/liferay-portal/pull/873/commits/0f562908bddb2da7230e3664c9e919d5d83ed724#diff-5f9a1cead69535423c815c23f96fc45737baee11a8a5604ae2c7d49f48e9436fR253 as a reference.